### PR TITLE
Remove ClassHierarchy and avoid duplicate work

### PR DESCRIPTION
**Pull request by jacob314**
Previously running the flutter tests with track-widget-creation took 4:40 seconds versus about 3:45 with the flag off.
Key is moving off the class hierarchy and avoiding duplicated work on already computed libraries.

### CODE REVIEW

**Comment by jacob314** [comment]
Fyi @sigmundch

File: flutter_kernel_transformers/lib/track_widget_constructor_locations.dart [hunk 1]
```diff
@@ -456,22 +441,17 @@ class WidgetCreatorTracker implements ProgramTransformer {
...
     final Set<Class> transformedClasses = Set<Class>.identity();
-    final Set<Library> librariesToTransform = Set<Library>.identity()
-      ..addAll(libraries);
+    final Set<Library> librariesToTransform = Set<Library>.identity();
 
     for (Library library in libraries) {
-      if (library.isExternal) {
-        continue;
+      if (!library.isExternal && !_isTransformed(library)) {
+        librariesToTransform.add(library);
```
Comment (askeksa-google) [inline flutter_kernel_transformers/lib/track_widget_constructor_locations.dart]: I am wondering if this check is necessary in the [newest SDK repo version](https://dart-review.googlesource.com/c/sdk/+/101987/6/pkg/kernel/lib/transformations/track_widget_constructor_locations.dart) of the transformer, which is given an explicit list of libraries to transform. For this reason, I have not brought the optimization over just yet.

Comment (jacob314) [reply flutter_kernel_transformers/lib/track_widget_constructor_locations.dart]: Sounds good. Hopefully it isn't needed.

**Comment by sigmundch** [comment]
just curious  - 4:40 was with the old implementation, 3:45 was with the transformer off, how long it takes now with the changes in this PR?

**Comment by jacob314** [comment]
Wow I left out the best part! Between 3:40 and 3:50 with the transformer ON with this fix.

**Comment by cbracken** [comment]
@jacob314 @askeksa-google what remains for us to land this change?

**Comment by jacob314** [comment]
Closing in favor of the other CL to achieve this goal already landed.

**Pull request closed without merging**